src/tox_ansible/tox_molecule_case.py

```diff
@@ -80,18 +83,41 @@ def get_dependencies(self) -> Iterable:
             dependencies.append("-r" + self.scenario.requirements)
         return dependencies
 
-    def get_name(self):
+    @lru_cache()
+    def get_name(self, fmt=""):
         """The name of this test case. The name is made up of various factors
         from the tox world, joined together by hyphens. Some of them are based
         on the role and scenario names. Others are based on factors such as
         the python version or ansible version.
 
         :return: The tox-friendly name of this test scenario"""
         scenario_path = self.scenario.directory.split(os.path.sep)
-        name_path = list(
-            filter(lambda x: x != "molecule" and "." not in x, scenario_path)
+        parts = list(filter(lambda x: x != "molecule" and "." not in x, scenario_path))
+
+        scenario = parts[-1]
+        role = ""
+        path = ""
+        if len(parts) >= 3:
+            path = "-".join(parts[:-2])
+            role = parts[-2]
```

Comment:
should we rename it to "parent_dir" instead and avoid referring to roles? 

I would also wish to be able to remove the `-default` which I find as spammy. If you have few minutes try to clone https://github.com/openstack/tripleo-ansible and run `tox -va` there, that is the big use-case I am trying to cover.

From my point of view a scenario named "default" has zero practical value and I do find as most useful "parent_dir + scenario_name" when that is not default, for default i would likely use only parent_dir alone. How to achieve this with a formatting option, not sure. Likely I will need to add another variable named "scenario_name_unless_default" or similar....

Reply:
Hmm, the naming bit I dunno. Do we want to maybe just call it "role" and put in documentation about the fact that it may or may not be the actual role name? (note to PR: we should document this option)

As for filtering out the "default", we can absolutely add in an exception to appending that name to the environment. The next best option would be either introducing a separate variable, as you suggest. If we went with a whole templating language we could be really flexible, but maybe just a "nondefault_scenario" variable?